src/ai/backend/manager/models/container_registry.py

```diff
@@ -0,0 +1,294 @@
+from __future__ import annotations
+
+import enum
+import logging
+import uuid
+from typing import TYPE_CHECKING, Any, Sequence
+
+import graphene
+import sqlalchemy as sa
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm.exc import NoResultFound
+
+from ai.backend.common.logging_utils import BraceStyleAdapter
+
+from ..defs import PASSWORD_PLACEHOLDER
+from .base import (
+    Base,
+    IDColumn,
+    StrEnumType,
+    privileged_mutation,
+    set_if_set,
+)
+from .gql_relay import AsyncNode
+from .user import UserRole
+
+if TYPE_CHECKING:
+    from .gql import GraphQueryContext
+
+log = BraceStyleAdapter(logging.getLogger(__spec__.name))  # type: ignore
+
+__all__: Sequence[str] = (
+    "ContainerRegistry",
+    "ContainerRegistryRow",
+    "CreateContainerRegistry",
+    "ModifyContainerRegistry",
+    "DeleteContainerRegistry",
+)
+
+
+class ContainerRegistryType(enum.StrEnum):
+    DOCKER = "docker"
+    HARBOR = "harbor"
+    HARBOR2 = "harbor2"
+    LOCAL = "local"
+
+
+class ContainerRegistryRow(Base):
+    __tablename__ = "container_registries"
+    id = IDColumn()
+    url = sa.Column("url", sa.String(length=512), index=True, nullable=False)
```

Comment:
Just a comment. We can use `ai.backend.manager.models.base.URLColumn` for this column. But it is UnicodeText type and updating existing "str to yarl.URL" codes would be bothersome.. so, it is up to you.